### DISCUSSION TITLE
Rich output from a Python CLI tool

### DISCUSSION THREAD

**Discussion by NikosAlexandris** [pawamoy/markdown-exec#42] [Q&A] [answered]
Unless I miss something, the [rich (console markup)](https://rich.readthedocs.io/en/stable/markup.html?highlight=print) print output from a Python (Typer-based) CLI tool is not preserved in the output. For example,

``` bash
cat printsomethingrich.py
```
``` pythpn
from rich import print
print(f'Something rich : [bold]bold[/bold] and [green]green[/green]')
```
and then 

```` markdown
``` bash exec="true" result="ansi" source="material-block"
python printsomethingrich.py
```
````

but the output

![2024-04-08T12:08:02,456069581+03:00](https://github.com/pawamoy/markdown-exec/assets/7046639/d55717c1-4e58-423e-8909-c2fcd9aa41ce)

isn't as expected in the terminal

![2024-04-08T12:08:51,186692363+03:00](https://github.com/pawamoy/markdown-exec/assets/7046639/739de429-c25a-4ab2-9f58-58175a251bee)

Is there a way to preserve the richness ?

**Comment by pawamoy** [comment, marked as answer]
It's probably because Rich detects its running in a subprocess and therefore deactivates colors. Try forcing them with this:

````md
``` bash exec="true" result="ansi" source="material-block"
export FORCE_COLOR=1  # markdown-exec: hide
python printsomethingrich.py
```
````

**Reply by NikosAlexandris** [reply]
That did it!

![2024-04-08T22:27:05,791081226+03:00](https://github.com/pawamoy/markdown-exec/assets/7046639/46a7af84-2f7f-40d4-9a09-f102e68e5c7f)


ps- Would you recommend a way to force this globally and not need to repeat in each codeblock ?

**Reply by pawamoy** [reply]
You can export the variable at the base level, in the shell process that runs MkDocs, and all subprocesses should inherit it :) (unless we specifically clear the environment when running a bash process but I don't think we do)

**Reply by NikosAlexandris** [reply]
Works, indeed.-
